docs/lofi-hifi-backends.md

```diff
@@ -0,0 +1,207 @@
+# Introduce Instructlab profiles managed via `ilab profile...` to run different key commands at different Fidelity levels
+
+This document describes adding different data generation, mixing, and training backends to `ilab` to enable higher fidelity training using the backend code. By higher fidelity we mean models that preform better, were trained on better hardware, off of larger data sets and using more intensive training techniques.
+
+Rather than augmenting the existing commands to have conflicting and confusing flags that the team will be forced to maintain, deprecate, and add onto, I am proposing we open the door to these new workflows via instructlab "profiles" and the `ilab profile` command group. 
+
+Currently all training is done via QLoRA or the like. Adding the following commands will enable higher fidelity training and introduce commands such as data mixing.
+
+This document focuses primarily on training, specifically different configuration types or "profiles" for `ilab model train`.
+
+## Key Component
+
+### Building off of the InstructLab Structure Redesign
+
+After <https://github.com/instructlab/instructlab/pull/990> is merged, ilab will use a parent -> child command structure. This proposal operates under that new structure.
+
+`ilab model train` will still be a single command offered to users. However, new "profiles" offered by the CLI will enable users to toggle what mode they are running instructlab in. These profiles will initially just have global options for training. Rather than adding dozens of flags to the CLI, storing them in "pre-baked" configs is a good first step.
+
+Eventually, we will want to enable a `ilab profile init` command that will allow users to initialize a custom profile based off of one of the pre-baked ones. This is not targeted for an upcoming release though. 
+
+### Immediate Goals
+
+For the near future, a single upper level profile that can be initialized via `ilab profile set <profile_name>`
+
+A profile can be either a json or yaml object that would look something like:
+
+```yaml
+profile:
+    train:
+        gpus: 0-4
+        epochs: 15
+        accelerator: deepspeed
+        ds_config: foobar.json
+```
+Eventually this profile would have settings for generation, eval, etc. But for immediate goals, hardcoded training settings is the MVP. Rather than having a `--config` option at the `ilab model train` level, storing the profile at the global level allows us to expand this idea to other ilab commands in the future. We need to be careful about how we introduce new concepts like this.
+
+For immediate releases I would introduce the idea of a profile, a command to set a specific profile, and hardcoded profiles that plug into key ilab commands namely training.
```

Comment:
This idea that a profile is not specific to training seems like a critical detail of this proposal.

My expectation was something very focused on training.

A global profile for everything is just the existing configuration file. You can have multiple configuration files if needed. I'm having a hard time deciding how this would end up being significantly different.

I think you're proposing a set of options that are supported in the config file but NOT via command line arguments. Is that right?

Honestly I think I'd rather just have all the command line arguments PLUS an extra one `--profile` that's shorthand for a set of defaults.

Reply:
also, this is the design doc for friday's design freeze so it needs to consider all commands being added